x-pack/plugins/observability/common/utils/formatters/size.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { memoize } from 'lodash';
+import { asDecimal } from './formatters';
+import { Maybe } from '../../typings';
+
+function asKilobytes(value: number) {
+  return `${asDecimal(value / 1000)} KB`;
+}
+
+function asMegabytes(value: number) {
+  return `${asDecimal(value / 1e6)} MB`;
+}
+
+function asGigabytes(value: number) {
+  return `${asDecimal(value / 1e9)} GB`;
+}
+
+function asTerabytes(value: number) {
+  return `${asDecimal(value / 1e12)} TB`;
+}
+
+function asBytes(value: number) {
+  return `${asDecimal(value)} B`;
+}
+
+const bailIfNumberInvalid = (cb: (val: number) => string) => {
+  return (val: Maybe<number>) => {
+    if (val === null || val === undefined || isNaN(val)) {
+      return '';
+    }
+    return cb(val);
+  };
+};
+
+export const getFixedByteFormatter = memoize((max: number) => {
+  const formatter = unmemoizedFixedByteFormatter(max);
+
+  return bailIfNumberInvalid(formatter);
+});
+
+export const asDynamicBytes = bailIfNumberInvalid((value: number) => {
+  return unmemoizedFixedByteFormatter(value)(value);
+});
+
+const unmemoizedFixedByteFormatter = (max: number) => {
+  if (max > 1e12) {
+    return asTerabytes;
+  }
+
+  if (max > 1e9) {
+    return asGigabytes;
+  }
+
+  if (max > 1e6) {
+    return asMegabytes;
+  }
+
+  if (max > 1000) {
+    return asKilobytes;
+  }
+
+  return asBytes;
+};
```

Comment:
I don't see this file as deleted. So it was just copied over from APM?

Reply:
Yeah :(

Reply:
(I tried copying one over, but they're all somehow connected, so ended up copying the whole folder and assuming we will start using those from APM in the near future).